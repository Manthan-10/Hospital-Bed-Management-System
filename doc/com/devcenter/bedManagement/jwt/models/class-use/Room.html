<!DOCTYPE HTML>
<html lang="en">
<head>
<!-- Generated by javadoc (18) on Mon May 30 12:53:08 IST 2022 -->
<title>Uses of Class com.devcenter.bedManagement.jwt.models.Room</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<meta name="dc.created" content="2022-05-30">
<meta name="description" content="use: package: com.devcenter.bedManagement.jwt.models, class: Room">
<meta name="generator" content="javadoc/ClassUseWriter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../script-dir/jquery-ui.min.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../jquery-ui.overrides.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../script-dir/jquery-3.5.1.min.js"></script>
<script type="text/javascript" src="../../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-use-page">
<script type="text/javascript">var pathtoroot = "../../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon"></span><span class="nav-bar-toggle-icon"></span><span class="nav-bar-toggle-icon"></span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Class</a></li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../index-files/index-1.html">Index</a></li>
<li><a href="../../../../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div class="nav-list-search"><label for="search-input">SEARCH:</label>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Class com.devcenter.bedManagement.jwt.models.Room" class="title">Uses of Class<br>com.devcenter.bedManagement.jwt.models.Room</h1>
</div>
<div class="caption"><span>Packages that use <a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#com.devcenter.bedManagement.jwt.controllers">com.devcenter.bedManagement.jwt.controllers</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#com.devcenter.bedManagement.jwt.models">com.devcenter.bedManagement.jwt.models</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#com.devcenter.bedManagement.jwt.repository">com.devcenter.bedManagement.jwt.repository</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#com.devcenter.bedManagement.jwt.security.services">com.devcenter.bedManagement.jwt.security.services</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<section class="class-uses">
<ul class="block-list">
<li>
<section class="detail" id="com.devcenter.bedManagement.jwt.controllers">
<h2>Uses of <a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a> in <a href="../../controllers/package-summary.html">com.devcenter.bedManagement.jwt.controllers</a></h2>
<div class="caption"><span>Methods in <a href="../../controllers/package-summary.html">com.devcenter.bedManagement.jwt.controllers</a> with parameters of type <a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">RoomController.</span><code><a href="../../controllers/RoomController.html#createRoom(com.devcenter.bedManagement.jwt.models.Room)" class="member-name-link">createRoom</a><wbr>(<a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a>&nbsp;room)</code></div>
<div class="col-last even-row-color">
<div class="block">Function createRoom is mapped to /api/test/room/addroom for adding rooms</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="com.devcenter.bedManagement.jwt.models">
<h2>Uses of <a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a> in <a href="../package-summary.html">com.devcenter.bedManagement.jwt.models</a></h2>
<div class="caption"><span>Methods in <a href="../package-summary.html">com.devcenter.bedManagement.jwt.models</a> that return <a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Bed.</span><code><a href="../Bed.html#getRoom()" class="member-name-link">getRoom</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">com.devcenter.bedManagement.jwt.models</a> with parameters of type <a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Bed.</span><code><a href="../Bed.html#setRoom(com.devcenter.bedManagement.jwt.models.Room)" class="member-name-link">setRoom</a><wbr>(<a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a>&nbsp;room)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../package-summary.html">com.devcenter.bedManagement.jwt.models</a> with parameters of type <a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../Bed.html#%3Cinit%3E(int,java.lang.String,com.devcenter.bedManagement.jwt.models.Room)" class="member-name-link">Bed</a><wbr>(int&nbsp;id,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;bedNo,
 <a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a>&nbsp;room)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="com.devcenter.bedManagement.jwt.repository">
<h2>Uses of <a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a> in <a href="../../repository/package-summary.html">com.devcenter.bedManagement.jwt.repository</a></h2>
<div class="caption"><span>Methods in <a href="../../repository/package-summary.html">com.devcenter.bedManagement.jwt.repository</a> that return types with arguments of type <a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">RoomRepository.</span><code><a href="../../repository/RoomRepository.html#getRoomsByFloor(com.devcenter.bedManagement.jwt.models.Floor)" class="member-name-link">getRoomsByFloor</a><wbr>(<a href="../Floor.html" title="class in com.devcenter.bedManagement.jwt.models">Floor</a>&nbsp;floor)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../../repository/package-summary.html">com.devcenter.bedManagement.jwt.repository</a> with parameters of type <a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Bed.html" title="class in com.devcenter.bedManagement.jwt.models">Bed</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">BedRepository.</span><code><a href="../../repository/BedRepository.html#getBedsByRoom(com.devcenter.bedManagement.jwt.models.Room)" class="member-name-link">getBedsByRoom</a><wbr>(<a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a>&nbsp;room)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="com.devcenter.bedManagement.jwt.security.services">
<h2>Uses of <a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a> in <a href="../../security/services/package-summary.html">com.devcenter.bedManagement.jwt.security.services</a></h2>
<div class="caption"><span>Methods in <a href="../../security/services/package-summary.html">com.devcenter.bedManagement.jwt.security.services</a> that return types with arguments of type <a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">RoomService.</span><code><a href="../../security/services/RoomService.html#getRooms()" class="member-name-link">getRooms</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Function for getting all rooms</div>
</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">RoomService.</span><code><a href="../../security/services/RoomService.html#getRoomsByFloor(int)" class="member-name-link">getRoomsByFloor</a><wbr>(int&nbsp;id)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../../security/services/package-summary.html">com.devcenter.bedManagement.jwt.security.services</a> with parameters of type <a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">RoomService.</span><code><a href="../../security/services/RoomService.html#createRoom(com.devcenter.bedManagement.jwt.models.Room)" class="member-name-link">createRoom</a><wbr>(<a href="../Room.html" title="class in com.devcenter.bedManagement.jwt.models">Room</a>&nbsp;room)</code></div>
<div class="col-last even-row-color">
<div class="block">Function for adding new rooms</div>
</div>
</div>
</section>
</li>
</ul>
</section>
</main>
</div>
</div>
</body>
</html>
