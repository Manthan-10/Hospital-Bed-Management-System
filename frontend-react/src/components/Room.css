.room{
  margin-left: 270px;
  background-color:white;
  margin-right:10px;
  min-height: 120vh;
}

.headings{
  margin-top:-40px;
  text-align:center;
}

.hospital-heading{
  font-size: 40px;
  font-weight: medium;
}

.floor-heading{
  font-size: 30px;
  margin-top: 30px;

}


.room-wrapper {
  display: flex;
  flex-direction: column;
  margin-left:80px;
  
  
}

.room-heading {
  margin-top: 40px;
  font-size: 20px;
  margin-bottom: 30px;
}


