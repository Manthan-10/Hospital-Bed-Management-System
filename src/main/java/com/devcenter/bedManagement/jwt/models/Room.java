package com.devcenter.bedManagement.jwt.models;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;

@Entity
@Table(name = "Room")
public class Room {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	
	private String roomNo;
	
	@ManyToOne(cascade = CascadeType.MERGE)
	@JoinColumn(name="Floor_id", referencedColumnName= "id")
	private Floor floor;
	
	@OneToMany(mappedBy="room", cascade = CascadeType.ALL)
	private List<Bed> beds;
	
	
	
	public Room() {
		super();
	}



	public Room(int id, String roomNo, Floor floor) {
		super();
		this.id = id;
		this.roomNo = roomNo;
		this.floor = floor;
	}



	public int getId() {
		return id;
	}



	public void setId(int id) {
		this.id = id;
	}



	public String getRoomNo() {
		return roomNo;
	}



	public void setRoomNo(String roomNo) {
		this.roomNo = roomNo;
	}



	public Floor getFloor() {
		return floor;
	}



	public void setFloor(Floor floor) {
		this.floor = floor;
	}

	
	
}
