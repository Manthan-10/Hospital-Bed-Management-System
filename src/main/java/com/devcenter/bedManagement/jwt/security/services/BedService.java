package com.devcenter.bedManagement.jwt.security.services;

import java.util.List;
import java.util.Optional;

import org.modelmapper.ModelMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.devcenter.bedManagement.jwt.models.Bed;
import com.devcenter.bedManagement.jwt.models.Room;
import com.devcenter.bedManagement.jwt.repository.BedRepository;
import com.devcenter.bedManagement.jwt.repository.RoomRepository;


/**
 * Service class
 * @author MN097789
 *
 */

@Service
public class BedService {
	
	@Autowired
	BedRepository bedRepository;
	
	@Autowired
	RoomRepository roomRepository;
	
	@Autowired
	private ModelMapper modelMapper;
	
	/**
	 * Function for adding new beds
	 * @param bed represents Bed object
	 * 	 
	 **/
	public void createBed(Bed bed) {
		 bedRepository.save(bed);
	}
	
	

	/**
	 * Function fetches all beds
	 * @return Returns List of all beds
	 */
	public List<Bed> getBeds() {
		return bedRepository.findAll(); 
	}
	
	
	/**
	 * Function fetches the details of all the beds for the given Room id 
	 * @param is id, which represents Room Id
	 * @return All Beds in that particular Room and its details
	 * 
	 */
	public List<Bed> getBedsByRoom(int id){
		
		Optional<Room> room = roomRepository.findById(id);
		List<Bed> beds = bedRepository.getBedsByRoom(this.modelMapper.map(room, Room.class));
		
		return beds;
		
	}
}
