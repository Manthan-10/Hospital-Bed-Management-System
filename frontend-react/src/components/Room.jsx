import React, { useEffect, useState } from "react";
import { useTranslation } from "react-i18next";
import BedService from "../services/bed.service";
import Bed from "./BedList";
import "./Room.css";

/**
 * This component displays all the Rooms in a particular Floor
 *
 * @param {fid} FloorId of that Floor
 */
const Room = ({ fid }) => {
  const [data, setRoomList] = useState([]);
  const { t } = useTranslation();

  /**
   * @param {fid} FloorId of that Floor
   * @return All Rooms and its details of that Floor
   */
  useEffect(() => {
    BedService.getRoomsByFid(fid)
      .then((response) => {
        console.log(response);
        setRoomList(response.data);
      })
      .catch((e) => {
        console.log(e);
      });
  }, []);

  return (
    <div className="room">
      <div className="headings">
        {data.map((room, index) =>
          index > -1 && index < 1 ? (
            <div key={room.id}>
              <p className="floor-heading">
                {room.floor.floorNo} {t("Floor")}
              </p>
            </div>
          ) : undefined
        )}
      </div>

      <div className="room-wrapper">
        {data.map((room) => (
          <div key={room.id}>
            <p className="room-heading">
              {t("Room")} {room.roomNo}
            </p>
            <Bed rid={room.id} />
          </div>
        ))}
      </div>
    </div>
  );
};

export default Room;
