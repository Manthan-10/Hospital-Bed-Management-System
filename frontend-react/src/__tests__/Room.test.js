import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Room from "../components/Room";
import Adapter from "enzyme-adapter-react-16";
import * as reactRouterDom from "react-router-dom";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import BedService from "../services/bed.service";

Enzyme.configure({ adapter: new Adapter() });

/**
 * Testing for UI of Login page
 * @author MN097789
 */

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  useSelector: jest.fn(),

  useNavigate: () => mockNavigate,
}));
const mockNavigateSpy = jest.spyOn(reactRouterDom, "useNavigate");

describe("Room UI Test Suite", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Room />);
  });

  test("Checks if it renders correctly", () => {
    const tree = renderer.create(<Room />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should check for divs", () => {
    wrapper = shallow(<Room />);
    expect(wrapper.length).toEqual(1);
  });
});
