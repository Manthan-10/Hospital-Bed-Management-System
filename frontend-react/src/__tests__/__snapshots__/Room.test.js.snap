// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room UI Test Suite Checks if it renders correctly 1`] = `
<div
  className="room"
>
  <div
    className="headings"
  />
  <div
    className="room-wrapper"
  />
</div>
`;
